src/Composer/Proxy.php

```diff
@@ -1,175 +0,0 @@
-<?php
-
-namespace Covex\Composer;
-
-use Composer\EventDispatcher\Event;
-use Composer\EventDispatcher\EventDispatcher;
-
-/**
- * @author Andrey F. Mindubaev
- */
-class Proxy
-{
-    /**
-     * @var object
-     */
-    private $object;
-
-    /**
-     * @var EventDispatcher
-     */
-    private $eventDispacher;
-
-    /**
-     * @var \ReflectionClass
-     */
-    private $reflection;
-
-    /**
-     * @var \ReflectionProperty[]
-     */
-    private $properties = [ ];
-
-    /**
-     * @var \ReflectionMethod[]
-     */
-    private $methods = [ ];
-
-    /**
-     * @var string
-     */
-    private $event;
-
-    /**
-     * @var array
-     */
-    private $subscribes = [ ];
-
-    public function __construct($object, EventDispatcher $eventDispatcher = null)
-    {
-        if (!is_object($object)) {
-            throw new \LogicException("Object must be object!");
-        }
-        $this->object = $object;
-        $this->reflection = new \ReflectionClass(get_class($object));
-
-        if (!is_null($eventDispatcher)) {
-            $this->setEventDispatcher($eventDispatcher);
-        }
-    }
-
-    /**
-     * @param string $event
-     * @param array  $subscribes
-     */
-    public function subscribe($event, array $subscribes)
-    {
-        if (is_null($this->eventDispacher)) {
-            throw new \RuntimeException("EventDispatcher must be set");
-        }
-        if (!is_string($event)) {
-            throw new \RuntimeException("EventName must be a string");
-        }
-        $this->event = $event;
-        $this->subscribes = $subscribes;
-    }
-
-    /**
-     * @param EventDispatcher $eventDispatcher
-     */
-    public function setEventDispatcher(EventDispatcher $eventDispatcher)
-    {
-        $this->eventDispacher = $eventDispatcher;
-    }
-
-    public function __call($name, $arguments)
-    {
-        if (in_array($name, $this->subscribes)) {
-            $eventName = "flex-" . strtolower($this->event) . "-" . $name;
-        } else {
-            $eventName = null;
-        }
-
-        $method = $this->getMethod($name);
-        if ($method) {
-            if (!is_null($eventName)) {
-                $preEvent = new Event("pre-" . $eventName, [ "arguments" => $arguments ]);
-                $this->eventDispacher->dispatch($preEvent->getName(), $preEvent);
```

Comment:
It was an experiment. There were an issue in `symfony/flex` about extending Flex. This fireing `pre-flex-configurator-install`, `post-flex-configurator-install` and `pre-flex-configurator-unconfigure` events, for example, for example. Plugin does not use it

`$flexPlugin->configurator` and `$flexPlugin->dowloader` are replaced by instanceof Proxy somewhere in `ExtraFlexPlugin::decorate`, and after that Flex plugin operates with Proxy.